e2e/cypress/integration/messaging/emoji_to_markdown_spec.js

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. 1. Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+function createMessages(message) {
+    const aliasLineMessage = 'newLineMessage';
+    const aliasLineMessageReplyInRHS = 'newLineMessageReplyInRHS';
+
+    cy.postMessage(message);
+    cy.getLastPostId().then((postId) => {
+        cy.get(`#postMessageText_${postId}`).as(aliasLineMessage);
+        cy.clickPostCommentIcon(postId);
+    });
+
+    cy.postMessageReplyInRHS(message + '{enter}');
+    cy.getLastPostId().then((postId) => {
+        cy.get(`#postMessageText_${postId}`).as(aliasLineMessageReplyInRHS);
+    });
+
+    return [aliasLineMessage, aliasLineMessageReplyInRHS];
+}
+
+function hasMarkdown(message) {
+    const aliases = createMessages(message);
+
+    aliases.forEach((alias) => {
+        cy.get('@' + alias).
+            children().should('have.class', 'post-code').
+            children('code').should('be.visible').contains(message.trim());
+    });
+}
+
+function hasNoMarkdown(message) {
+    const aliases = createMessages(message);
+
+    aliases.forEach((alias) => {
+        cy.get('@' + alias).
+            children().should('not.have.class', 'post-code').
+            children('code').should('have.length', 0);
+    });
+}
+
+describe('Messaging', () => {
+    before(() => {
+        cy.apiLogin('user-1');
+        cy.visit('/');
+    });
+
+    it('MM-47736 - Emojis preceded by 4 or more spaces are treated as Markdown', () => {
```

Comment:
Last one as I corrected the test ID. Please change to
```suggestion
    it('M17446 - Emojis preceded by 4 or more spaces are treated as Markdown', () => {
```

Reply:
outdated, please have another look.